model.py

```diff
@@ -221,18 +197,19 @@ def _load(path):
             "optimizer saved in dygraph mode is not usable in static graph"
 
         if self._executor is not None:
-           self._load_optimizer(optim_state)
+            self._load_optimizer(optim_state)
         else:
-           self._lazy_load_optimizer = optim_state
+            self._lazy_load_optimizer = optim_state
```

Comment:
if program construction happens in `prepare`,  these lazy loading mechanism are just cruft, and should be removed for simplification.

Reply:
As above reply.